### PR TITLE
Added java field type

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: queries/java.scm [hunk 1]
```diff
@@ -326,3 +326,32 @@
...
+(class_body
+  .
+  "{" @type.iteration.start.endOf
+  "}" @type.iteration.end.startOf
+  .
+)
+
+;;!! public Map<int, int> foo;
+;;!             ^^^  ^^^
+(integral_type) @type
```
Comment (pokey) [inline queries/java.scm:348]: What is `integral_type` generally?

Comment (AndreasArvidsson) [reply queries/java.scm]: Not what we want :p
It apparently only consists of the basic types. I have updated the pattern now.

Comment (pokey) [reply queries/java.scm]: ok cool yeah that's what I had assumed based on its name 😅

File: queries/java.scm [hunk 2]
```diff
@@ -326,3 +326,34 @@
...
+  "{" @type.iteration.start.endOf
+  "}" @type.iteration.end.startOf
+  .
+)
+
+;;!! public Map<int, int> foo;
+;;!             ^^^  ^^^
+(type_arguments
+  (_) @type
+)
```
Comment (pokey) [inline queries/java.scm:350]: shall we add a facet for this?

Comment (AndreasArvidsson) [reply queries/java.scm:350]: Probably. What should we call that? `type.typeArguments`?

Comment (pokey) [reply queries/java.scm:350]: Works for me

File: queries/java.scm [hunk 3]
```diff
@@ -326,3 +326,34 @@
...
+) @_.domain
+
+;;!! class MyClass { }
+;;!                 ^
+(class_body
+  .
+  "{" @type.iteration.start.endOf
+  "}" @type.iteration.end.startOf
+  .
+)
```
Comment (pokey) [inline queries/java.scm:344]: We should prob have facets for this iteration scope and the one below as well

File: queries/java.scm (hunk 2 above)
Comment (AndreasArvidsson) [reply queries/java.scm:350]: done

File: packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts [hunk 4]
```diff
@@ -417,4 +423,14 @@ export const scopeSupportFacetInfos: Record<
     description: "A type cast",
     scopeType: "type",
   },
+  "type.typeArgument": {
+    description: "Type argument to a collection type",
```
Comment (pokey) [inline packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts:427]:
```suggestion
    description: "Type argument to a generic type",
```

File: packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts [hunk 5]
```diff
@@ -417,4 +423,14 @@ export const scopeSupportFacetInfos: Record<
...
+    scopeType: "type",
+  },
+  "type.typeArgument.iteration": {
+    description:
+      "Iteration scope for type argument to a collection type; Should be the list of type arguments",
```
Comment (pokey) [inline packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts:432]:
```suggestion
      "Iteration scope for type argument to a generic type; Should be the list of type arguments",
```

**Review by pokey — APPROVED** [review]

File: packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts [hunk 6]
```diff
@@ -417,4 +423,14 @@ export const scopeSupportFacetInfos: Record<
     description: "A type cast",
     scopeType: "type",
   },
+  "type.typeArgument": {
+    description: "Type argument to a generic type",
```
Comment (pokey) [inline packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts:427]:
```suggestion
    description: "Type argument to a generic / parametrized type",
```

File: packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts [hunk 7]
```diff
@@ -417,4 +423,14 @@ export const scopeSupportFacetInfos: Record<
...
+    scopeType: "type",
+  },
+  "type.typeArgument.iteration": {
+    description:
+      "Iteration scope for type argument to a generic type; Should be the list of type arguments",
```
Comment (pokey) [inline packages/common/src/scopeSupportFacets/scopeSupportFacetInfos.ts:432]:
```suggestion
      "Iteration scope for type argument to a generic / parametrized type; Should be the list of type arguments",
```

**Pull request merged**